nixos/modules/services/monitoring/prometheus/default.nix

```
@@ -62,15 +62,16 @@ let
     promtoolCheck "check config ${lib.optionalString (cfg.checkConfig == "syntax-only") "--syntax-only"}" "prometheus.yml" yml;
 
   cmdlineArgs = cfg.extraFlags ++ [
-    "--storage.tsdb.path=${workingDir}/data/"
     "--config.file=${
       if cfg.enableReload
       then "/etc/prometheus/prometheus.yaml"
       else prometheusYml
     }"
     "--web.listen-address=${cfg.listenAddress}:${builtins.toString cfg.port}"
-    "--alertmanager.notification-queue-capacity=${toString cfg.alertmanagerNotificationQueueCapacity}"
-  ] ++ optional (cfg.webExternalUrl != null) "--web.external-url=${cfg.webExternalUrl}"
+  ] ++ optional (!(cfg.enableAgentMode)) "--alertmanager.notification-queue-capacity=${toString cfg.alertmanagerNotificationQueueCapacity }"
+    ++ optional (!(cfg.enableAgentMode)) "--storage.tsdb.path=${workingDir}/data/"
+    ++ optional (cfg.enableAgentMode) "--enable-feature=agent"
```

```suggestion
  ] ++ ( 
    if (cfg.enableAgentMode) then [
      "--enable-feature=agent" 
    ] else [ 
       "--alertmanager.notification-queue-capacity=${toString cfg.alertmanagerNotificationQueueCapacity }" 
       "--storage.tsdb.path=${workingDir}/data/"
    ])
```

(Ignore my indentation)